streaming/src/main/scala/org/apache/spark/streaming/ui/StreamingTab.scala

```diff
@@ -26,37 +25,24 @@ import org.apache.spark.ui.{SparkUI, SparkUITab}
  * Spark Web UI tab that shows statistics of a streaming job.
  * This assumes the given SparkContext has enabled its SparkUI.
  */
-private[spark] class StreamingTab(val ssc: StreamingContext)
-  extends SparkUITab(StreamingTab.getSparkUI(ssc), "streaming") with Logging {
-
-  import StreamingTab._
+private[spark] class StreamingTab(val ssc: StreamingContext, sparkUI: SparkUI)
```

Comment:
Is it necessary to plumb through the reference to `SparkUI` vs getting it how it's done now? If it's just for tidiness, I think that's probably OK too.

Reply:
It's not necessary, but i believe it's a bit simpler and also consistent with other ui tabs.